src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -14,18 +14,44 @@ interface DropdownMenuItemProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   className?: string;
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  // To select a certain option in the menu without having to use document.querySelector
+  customRef?: React.RefObject<HTMLDivElement>;
+  onActivate?: (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) => void;
 }
 
 export const DropdownMenuDivider: React.FC = () => <div className="tk-dropdown-menu-divider"></div>
 
-export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, ...rest}: DropdownMenuItemProps) => {
+export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, customRef, onActivate, ...rest}: DropdownMenuItemProps) => {
   const classes = classNames(
     'tk-dropdown-menu__item',
     className,
   )
 
+  const focusNextOption = (current: HTMLDivElement, direction: number) => {
+    const options = (Array.from(current.parentElement.getElementsByClassName('tk-dropdown-menu__item'))) as HTMLDivElement[];
```

Comment:
Is it possible to use `current.nextElementSibling` or `current.prevElementSibling`?

Those will give you exactly the elements you are looking for, right?

Selecting based on classes means you select based on the styling. If the way the element is styled changes, the navigation also breaks. Adding wrapper elements, or adding child elements with the same class will also confuse the navigation code. Selecting by class name gives you what you need right now, but there are many future code changes that are completely unrelated to keyboard navigation that would break your code.

I think it would be better to write the navigation code completely independent from styling.

Reply:
The problem with doing `current.nextElementSibling` or `current.prevElementSibling` is that when you reach the beginning or end of the list, it won't jump to the first/last option as there is no previous or next sibling anymore, which is not the desirable experience. You would expect it to focus the first option when you press arrowDown on the last option, and the other way around.

Reply:
Could you use `current.nextElementSibling` and `current.previousElementSibling`, and if they are undefined use the `current.parent.lastChild` and `current.parent.firstChild`. Shouldn't that work?

Reply:
You can have other non-menu options as well, such as a divider. and you don't want to focus those.

Reply:
Okay, tricky..